forms-flow-api/src/formsflow_api/models/application.py

```diff
@@ -295,6 +295,23 @@ def find_by_form_names(  # pylint: disable=too-many-arguments, too-many-position
         pagination = query.paginate(page=page_no, per_page=limit, error_out=False)
         return pagination.items, total_count
 
+    @classmethod
+    def find_applications_count_by_parent_form_id_user(
+        cls, parent_form_id, user_name, tenant
+    ):
+        """Fetch application count based on parent_form_id and user who submitted the application."""
+        count_query = (
+            db.session.query(func.count(Application.id))
+            .join(FormProcessMapper, cls.form_process_mapper_id == FormProcessMapper.id)
+            .filter(
+                FormProcessMapper.parent_form_id == parent_form_id,
+                FormProcessMapper.tenant == tenant,
+                cls.created_by == user_name,
+                cls.application_status != DRAFT_APPLICATION_STATUS,
```

Comment:
@auslin-aot As per our discussion, we would be adding another flag to the table and use it to check for drafts rather than relying on status right ?

Reply:
Yes. That is done as part of another ticket(https://aottech.atlassian.net/browse/FWF-3749) & PR
https://github.com/AOT-Technologies/forms-flow-ai/pull/2533. Once that is done I will do the changes in this file too. Wether it's ok?